src/operation.rs

```diff
@@ -36,38 +37,49 @@ fn deserialize_operation_type<'de, D>(de: D) -> Result<OperationType, D::Error>
 }
 
 impl Operation {
-    pub fn apply(&self, old_lines: &[Line], old_line_index: u64, new_lines: &mut Vec<Line>) -> u64 {
+    pub fn apply(&self, old_lines: &[Line], old_ix: u64, new_lines: &mut Vec<Line>) -> Result<u64> {
+        // FIXME: this method panics if we don't check old_lines indices access.
```

Comment:
[9c8afa](https://github.com/little-dude/xi-tui/commit/9c8afa1d98fac4271cbe3038f21b82777638d73e) addresses this